frame/offences/benchmarking/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Offences pallet benchmarking.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use sp_std::prelude::*;
+use sp_std::vec;
+
+use frame_system::RawOrigin;
+use frame_benchmarking::{benchmarks, account};
+use frame_support::traits::{Currency, OnInitialize};
+
+use sp_runtime::{Perbill, traits::{Convert, StaticLookup}};
+use sp_staking::offence::ReportOffence;
+
+use pallet_im_online::{Trait as ImOnlineTrait, UnresponsivenessOffence};
+use pallet_offences::{Trait as OffencesTrait, Module as OffencesModule};
+use pallet_staking::{
+	Module as Staking, Trait as StakingTrait, RewardDestination, ValidatorPrefs,
+	Exposure, IndividualExposure, ElectionStatus
+};
+use pallet_session::Trait as SessionTrait;
+use pallet_session::historical::{Trait as HistoricalTrait, IdentificationTuple};
+
+const SEED: u32 = 0;
+
+
+const MAX_USERS: u32 = 1000;
+const MAX_REPORTERS: u32 = 100;
+const MAX_OFFENDERS: u32 = 100;
+const MAX_NOMINATORS: u32 = 100;
+
+pub struct Module<T: Trait>(OffencesModule<T>);
+
+pub trait Trait: SessionTrait + StakingTrait + OffencesTrait + ImOnlineTrait {}
+
+fn create_offender<T: Trait>(n: u32, nominators: u32) -> Result<T::AccountId, &'static str> {
+	let stash: T::AccountId = account("stash", n, SEED);
+	let controller: T::AccountId = account("controller", n, SEED);
+	let controller_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(controller.clone());
+	let reward_destination = RewardDestination::Staked;
+	let amount = T::Currency::minimum_balance();
+
+	Staking::<T>::bond(
```

Comment:
@shawntabrizi such functionality can also be ideally shared in the long term future where we'd have one utility function `bond(...)` for all the benchmarks. But a story for another day.